### DISCUSSION TITLE
Creating the virtual environment fails

### DISCUSSION THREAD

**Discussion by OlegHahm** [benfogle/crossenv#144] [Q&A / Support] [answered]
After installing crossenv (`sudo pip3.5 install crossenv`) I tried to create the virtual environment but end up with:
```
python3.5 -m crossenv ~/python-host-3.5/usr/bin/python3 cross_venv                                                                                                                               [11:39:43][1]
WARNING: CC is a compound command (arm-tdx-linux-gnueabi-gcc  -march=armv7-a -mthumb -mfpu=neon -mfloat-abi=hard -DNDEBUG -fno-inline -fstack-protector-strong  -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security )
WARNING: This can cause issues for modules that don't expect it.
WARNING: Consider setting CC='arm-tdx-linux-gnueabi-gcc' and CFLAGS='-march=armv7-a -mthumb -mfpu=neon -mfloat-abi=hard -DNDEBUG -fno-inline -fstack-protector-strong -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security'
WARNING: CXX is a compound command (arm-tdx-linux-gnueabi-g++  -march=armv7-a -mthumb -mfpu=neon -mfloat-abi=hard -DNDEBUG -fno-inline -fstack-protector-strong  -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security )
WARNING: This can cause issues for modules that don't expect it.
WARNING: Consider setting CXX='arm-tdx-linux-gnueabi-g++' and CXXFLAGS='-march=armv7-a -mthumb -mfpu=neon -mfloat-abi=hard -DNDEBUG -fno-inline -fstack-protector-strong -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security'
ERROR: Cannot find Makefile
```
I understand the warnings (but tend to ignore them for now), but I do not understand the error. Which Makefile is crossenv looking for?

**Comment by benfogle** [comment, marked as answer]
Crossenv is trying to gather information about the builld-python installation from the Makefile that was used to build Python. This is produced after you run `configure` and installed in the Python library dir (e.g., the system default Python might put it in  `/usr/lib/python3.8/config-3.8-x86_64-linux-gnu/Makefile`).

I'm not entirely sure what the issue would be here, but here are some suggestions. First, I always recommend building your own build-python rather than using the system version. While I would also recommend installing your build-python to a local directory so that you don't need to use sudo, ensure that `sudo python3.5` and `python3.5` are running the same command: sudo doesn't use the user's $PATH by default.
